### ISSUE TITLE
Ex - closures (map, flatMap etc.) - handling incorrect

### ISSUE THREAD

**Issue by Sciss** [Sciss/Lucre#23]
In GitLab by @Sciss on Oct 28, 2019, 24:10

The following illustrates the problem:

```scala
// Widget
val cb = CheckBox("Enable")
val i1 = IntField()
val valueOpt = If (cb.selected()) Then Some(i1.value()) Else None
val i2 = IntField()
val valueOpt2 = valueOpt.map(_ + i2.value())
val lb = Label(valueOpt2.toStr)
lb.hAlign = Align.Center

BorderPanel(
  north  = FlowPanel(cb, i1, i2),
  center = lb
)
```

Changing the value of `i2` has no effect on `valueOpt2`.
The reason being that `ExpandedMapExOption` does not listen to
the expanded `fun`. Indeed, this would only explain by changing `i2` is not reflected
in `valueOpt2`, but it doesn't explain by disabling and enabling the check box does
not take `i2` into account at all (it has a permanently cached value of zero).

----

This problem will be much more difficult when mapping from `Seq`, because here we
need `It` and evaluate the closure multiple times. Therefore, we cannot have "an instance"
of the expanded closure and monitor that for changes. Instead we would need to detect all
environmental expressions referred to from the closure (here `i2.value()`), 
and monitor their changes "proactively"...

**Comment by Sciss** [comment]
In GitLab by @Sciss on Oct 28, 2019, 24:11

mentioned in commit 332848618c395cfab609273f91ab4f2cb832ae6a

**Comment by Sciss** [comment]
In GitLab by @Sciss on Nov 6, 2019, 24:00

mentioned in commit add3af95e18631ae2f1c30405c62be1cdbc590f2

**Comment by Sciss** [comment]
In GitLab by @Sciss on Nov 6, 2019, 24:01

closed via commit add3af95e18631ae2f1c30405c62be1cdbc590f2
